### PR TITLE
[Android] Add support to Map type parameter

### PR DESCRIPTION
**Pull request by xzhan96**
Currently, reflection does not support Collection type parameter
or return value, such case happens in some API upgrade.

BUG=XWALK-5631

### CODE REVIEW

**Comment by xzhan96** [comment]
@axinging @lincsoon @XiaosongWei @wuhengzhi

**Comment by xzhan96** [comment]
@halton please help to review, thanks.

**Comment by lincsoon** [comment]
LGTM

**Comment by xzhan96** [comment]
@wuhengzhi 
1. because python need difference separator, so Map<String, String> has different regular rule HashMap<String>
2. Generic type includes Collection type, and other format, so I describe it as above.

File: tools/reflection_generator/java_method.py [hunk 1]
```diff
@@ -10,7 +10,11 @@
 from string import Template
 
 def ConvertClassExpressionToClassType(class_name):
+  """ Turn "Map<String, String>" to Map.class. """
+  generic_re = re.compile('[a-zA-z0-9]+(\<[a-zA-Z0-9]+,\s[a-zA-z0-9]+\>)')
   """ Turn "final HashMap<String>" to HashMap.class. """
```
Comment (XiaosongWei) [inline tools/reflection_generator/java_method.py]: This comment should go with the original statement. others LGTM

**Comment by xzhan96** [comment]
@lincsoon could you help to merge ?

**Pull request merged**